### ISSUE TITLE
GLB fails to import textures

### ISSUE THREAD

**Issue by Wavesonics** [godotengine/godot#44027]
**Godot version:**
4.0

**OS/device including version:**
Win64

**Issue description:**
This GLB imports it's geometry correctly once #44023 is merged.

However it's textures fail to load:

It is all black: (4.0)
![Screenshot 2020-12-01 154828](https://user-images.githubusercontent.com/406473/100810338-20790580-33ed-11eb-9e63-d1ea23eceeac.jpg)

It should look like this: (3.2)
![Screenshot 2020-12-01 155127](https://user-images.githubusercontent.com/406473/100810348-27a01380-33ed-11eb-8265-1947522d51be.jpg)

These errors show up in the terminal: (4.0)
```
ERROR: 1 shaders of type CanvasSdfShaderRD were never freed
   at: ShaderRD::~ShaderRD (servers\rendering\rasterizer_rd\shader_rd.cpp:470)
ERROR: Invalid Object id '25367151136', can't call method.
   at: (D:\Code\External\godot\core/object/callable_method_pointer.h:95)

ERROR: Condition "p_size < 0" is true. Returning: ERR_INVALID_PARAMETER
   at: CowData<struct Vector2>::resize (D:\Code\External\godot\core/templates/cowdata.h:252)
ERROR: Cannot open file 'res://.import/cop_car.glb-d1da082c95816014f295a29160723f8f.scn'.
   at: (core\io\resource_format_binary.cpp:1030)
Adding mesh..
Generate..
Final texture size: 134.000000,134.000000 - max 133.500000,133.500000
ERROR: Another resource is loaded from path 'res://cop_car/police_car.mesh' (possible cyclic resource inclusion).
   at: (core\io\resource.cpp:79)
ERROR: Another resource is loaded from path 'res://cop_car/police_car.mesh' (possible cyclic resource inclusion).
   at: (core\io\resource.cpp:79)
ERROR: Another resource is loaded from path 'res://cop_car/police_car.mesh' (possible cyclic resource inclusion).
   at: (core\io\resource.cpp:79)
ERROR: Another resource is loaded from path 'res://cop_car/police_car.mesh' (possible cyclic resource inclusion).
   at: (core\io\resource.cpp:79)
ERROR: Another resource is loaded from path 'res://cop_car/police_car.mesh' (possible cyclic resource inclusion).
   at: (core\io\resource.cpp:79)
ERROR: Another resource is loaded from path 'res://cop_car/police_car.mesh' (possible cyclic resource inclusion).
   at: (core\io\resource.cpp:79)
ERROR: Another resource is loaded from path 'res://cop_car/police_car.mesh' (possible cyclic resource inclusion).
   at: (core\io\resource.cpp:79)
ERROR: Another resource is loaded from path 'res://cop_car/police_car.mesh' (possible cyclic resource inclusion).
   at: (core\io\resource.cpp:79)
ERROR: Another resource is loaded from path 'res://cop_car/police_car.mesh' (possible cyclic resource inclusion).
   at: (core\io\resource.cpp:79)
ERROR: Another resource is loaded from path 'res://cop_car/police_car.mesh' (possible cyclic resource inclusion).
   at: (core\io\resource.cpp:79)
ERROR: Texture (binding: 0, index 0) is not a valid texture.
   at: (drivers\vulkan\rendering_device_vulkan.cpp:4680)
ERROR: Condition "!uniform_set" is true.
   at: RenderingDeviceVulkan::draw_list_bind_uniform_set (drivers\vulkan\rendering_device_vulkan.cpp:6143)
ERROR: Uniforms were never supplied for set (3) at the time of drawing, which are required by the pipeline
   at: (drivers\vulkan\rendering_device_vulkan.cpp:6285)
ERROR: Texture (binding: 0, index 0) is not a valid texture.
   at: (drivers\vulkan\rendering_device_vulkan.cpp:4680)
ERROR: Condition "!uniform_set" is true.
   at: RenderingDeviceVulkan::draw_list_bind_uniform_set (drivers\vulkan\rendering_device_vulkan.cpp:6143)
ERROR: Uniforms were never supplied for set (3) at the time of drawing, which are required by the pipeline
   at: (drivers\vulkan\rendering_device_vulkan.cpp:6285)
ERROR: Texture (binding: 0, index 0) is not a valid texture.
   at: (drivers\vulkan\rendering_device_vulkan.cpp:4680)
ERROR: Condition "!uniform_set" is true.
   at: RenderingDeviceVulkan::draw_list_bind_uniform_set (drivers\vulkan\rendering_device_vulkan.cpp:6143)
ERROR: Uniforms were never supplied for set (3) at the time of drawing, which are required by the pipeline
   at: (drivers\vulkan\rendering_device_vulkan.cpp:6285)
ERROR: Texture (binding: 0, index 0) is not a valid texture.
   at: (drivers\vulkan\rendering_device_vulkan.cpp:4680)
ERROR: Condition "!uniform_set" is true.
   at: RenderingDeviceVulkan::draw_list_bind_uniform_set (drivers\vulkan\rendering_device_vulkan.cpp:6143)
ERROR: Uniforms were never supplied for set (3) at the time of drawing, which are required by the pipeline
   at: (drivers\vulkan\rendering_device_vulkan.cpp:6285)
ERROR: Texture (binding: 0, index 0) is not a valid texture.
   at: (drivers\vulkan\rendering_device_vulkan.cpp:4680)
ERROR: Condition "!uniform_set" is true.
   at: RenderingDeviceVulkan::draw_list_bind_uniform_set (drivers\vulkan\rendering_device_vulkan.cpp:6143)
ERROR: Uniforms were never supplied for set (3) at the time of drawing, which are required by the pipeline
   at: (drivers\vulkan\rendering_device_vulkan.cpp:6285)
ERROR: Texture (binding: 0, index 0) is not a valid texture.
   at: (drivers\vulkan\rendering_device_vulkan.cpp:4680)
ERROR: Condition "!uniform_set" is true.
   at: RenderingDeviceVulkan::draw_list_bind_uniform_set (drivers\vulkan\rendering_device_vulkan.cpp:6143)
ERROR: Uniforms were never supplied for set (3) at the time of drawing, which are required by the pipeline
   at: (drivers\vulkan\rendering_device_vulkan.cpp:6285)
ERROR: Texture (binding: 0, index 0) is not a valid texture.
   at: (drivers\vulkan\rendering_device_vulkan.cpp:4680)
ERROR: Condition "!uniform_set" is true.
   at: RenderingDeviceVulkan::draw_list_bind_uniform_set (drivers\vulkan\rendering_device_vulkan.cpp:6143)
ERROR: Uniforms were never supplied for set (3) at the time of drawing, which are required by the pipeline
   at: (drivers\vulkan\rendering_device_vulkan.cpp:6285)
ERROR: Texture (binding: 0, index 0) is not a valid texture.
   at: (drivers\vulkan\rendering_device_vulkan.cpp:4680)
ERROR: Condition "!uniform_set" is true.
   at: RenderingDeviceVulkan::draw_list_bind_uniform_set (drivers\vulkan\rendering_device_vulkan.cpp:6143)
ERROR: Uniforms were never supplied for set (3) at the time of drawing, which are required by the pipeline
   at: (drivers\vulkan\rendering_device_vulkan.cpp:6285)
ERROR: Texture (binding: 0, index 0) is not a valid texture.
   at: (drivers\vulkan\rendering_device_vulkan.cpp:4680)
ERROR: Condition "!uniform_set" is true.
   at: RenderingDeviceVulkan::draw_list_bind_uniform_set (drivers\vulkan\rendering_device_vulkan.cpp:6143)
ERROR: Uniforms were never supplied for set (3) at the time of drawing, which are required by the pipeline
   at: (drivers\vulkan\rendering_device_vulkan.cpp:6285)
ERROR: Texture (binding: 0, index 0) is not a valid texture.
   at: (drivers\vulkan\rendering_device_vulkan.cpp:4680)
ERROR: Condition "!uniform_set" is true.
   at: RenderingDeviceVulkan::draw_list_bind_uniform_set (drivers\vulkan\rendering_device_vulkan.cpp:6143)
ERROR: Uniforms were never supplied for set (3) at the time of drawing, which are required by the pipeline
   at: (drivers\vulkan\rendering_device_vulkan.cpp:6285)
ERROR: Texture (binding: 0, index 0) is not a valid texture.
   at: (drivers\vulkan\rendering_device_vulkan.cpp:4680)
ERROR: Condition "!uniform_set" is true.
   at: RenderingDeviceVulkan::draw_list_bind_uniform_set (drivers\vulkan\rendering_device_vulkan.cpp:6143)
ERROR: Uniforms were never supplied for set (3) at the time of drawing, which are required by the pipeline
   at: (drivers\vulkan\rendering_device_vulkan.cpp:6285)
ERROR: Texture (binding: 0, index 0) is not a valid texture.
   at: (drivers\vulkan\rendering_device_vulkan.cpp:4680)
ERROR: Condition "!uniform_set" is true.
   at: RenderingDeviceVulkan::draw_list_bind_uniform_set (drivers\vulkan\rendering_device_vulkan.cpp:6143)
ERROR: Uniforms were never supplied for set (3) at the time of drawing, which are required by the pipeline
   at: (drivers\vulkan\rendering_device_vulkan.cpp:6285)
ERROR: Texture (binding: 0, index 0) is not a valid texture.
   at: (drivers\vulkan\rendering_device_vulkan.cpp:4680)
ERROR: Condition "!uniform_set" is true.
   at: RenderingDeviceVulkan::draw_list_bind_uniform_set (drivers\vulkan\rendering_device_vulkan.cpp:6143)
ERROR: Uniforms were never supplied for set (3) at the time of drawing, which are required by the pipeline
   at: (drivers\vulkan\rendering_device_vulkan.cpp:6285)
```

**Steps to reproduce:**
Load the sample project in master, it will import the `cop_car.glb` file. Then in Godot, open the model to view it.

**Minimal reproduction project:**
[ImportCrash.zip](https://github.com/godotengine/godot/files/5626596/ImportCrash.zip)

**Comment by fire** [comment]
Is this fixed?

**Comment by Wavesonics** [comment]
No, this appears to be an unrelated issue that only exists in 4.0

**Comment by fire** [comment]
A entirely black scene should give a black shading. Are you sure you have lighting?

**Comment by Blackiris** [comment]
Just tried again to open the reproduction project, with one of the latest master build (14cc1df1d8921600758bb40d27dde1200f109414)  and I don't encounter the issue.
![image](https://user-images.githubusercontent.com/440909/125111110-c8725200-e0e5-11eb-867e-f7d6fc495ee1.png)


Could you confirm if it works now for you?

**Comment by fire** [comment]
As this is confirmed fixed, we close this issue.

If it is still relevant in the way it was described and discussed above, please comment to ask for it to be reevaluated. If it is only partly relevant, or if the original issue has changed, it would be better to open a new issue (potentially referring to this one) focused on the current state.
